pkg/preparer/orchestrate.go

```diff
@@ -155,7 +155,39 @@ func (p *Preparer) verifySignature(manifest pods.PodManifest, logger logging.Log
 
 	signer, err := manifest.Signer(p.keyring)
 	if signer != nil {
-		logger.WithField("signer_key", signer.PrimaryKey.KeyIdShortString()).Debugln("Resolved manifest signature")
+		signerId := signer.PrimaryKey.KeyIdShortString()
+		logger.WithField("signer_key", signerId).Debugln("Resolved manifest signature")
+
+		// Hmm, some hacks here.
+		if manifest.Id == "p2-preparer" {
+			raw, ok := manifest.Config["authorized_deployers"]
```

Comment:
Nooooooo!!!

This code says "when deploying p2-preparer, make sure the new manifest is signed by someone that the new manifest lists as authorized". That's no good.

The preparer should, on startup, read the authorized_deployers from its config, and store it somewhere.
Then it should check any p2-preparer manifests against that stored set of authorized_deployers.
That mechanism implements the desired behavior of "when deploying p2-preparer, make sure the new manifest is signed by someone that the OLD manifest lists as authorized"